src/Blocks/Bookmark.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Notion\Blocks;
+
+use Notion\Common\RichText;
+
+/**
+ * Bookmark block
+ *
+ * @psalm-import-type BlockJson from Block
+ * @psalm-import-type RichTextJson from \Notion\Common\RichText
+ *
+ * @psalm-type BookmarkJson = array{
+ *      bookmark: array{
+ *          url: string,
+ *          caption: RichTextJson[],
+ *      },
+ * }
+ */
+class Bookmark implements BlockInterface
+{
+    private const TYPE = Block::TYPE_BOOKMARK;
+
+    private Block $block;
+
+    private string $url;
+
+    /** @var \Notion\Common\RichText[] */
+    private array $caption;
+
+    /** @param \Notion\Common\RichText[] $caption */
+    private function __construct(Block $block, string $url, array $caption) {
+        if (!$block->isBookmark()) {
+            throw new \Exception("Block must be of type " . self::TYPE);
+        }
+
+        $this->block = $block;
+        $this->url = $url;
+        $this->caption = $caption;
+    }
+
+    /**
+     * Create a bookmark from a URL
+     */
+    public static function create(string $url): self
+    {
+        $block = Block::create(self::TYPE);
+
+        return new self($block, $url, []);
+    }
+
+    /** @internal */
+    public static function fromArray(array $array): self
+    {
+        /** @psalm-var BlockJson $array */
+        $block = Block::fromArray($array);
+
+        /** @psalm-var BookmarkJson $array */
+        $url = $array[self::TYPE]["url"];
+
+        $caption = array_map(fn($t) => RichText::fromArray($t), $array[self::TYPE]["caption"]);
+
+        return new self($block, $url, $caption);
+    }
+
+    /** @internal */
+    public function toArray(): array
```

Comment:
Consider eating your own dogfood here, and declaring `/** @return BookmarkJson */`

Reply:
Ticket already created for similar comment.